File: src/main/java/yuu/rpg/Join.java
package yuu.rpg;

import org.bukkit.Location;
import org.bukkit.Material;
import org.bukkit.World;
import org.bukkit.configuration.file.FileConfiguration;
import org.bukkit.entity.Player;
import org.bukkit.entity.Villager;
import org.bukkit.event.EventHandler;
import org.bukkit.event.EventPriority;
import org.bukkit.event.Listener;
import org.bukkit.event.player.PlayerJoinEvent;
import org.bukkit.inventory.ItemStack;
import org.bukkit.inventory.meta.BookMeta;

import java.util.UUID;

public class Join implements Listener {

    private RPG plugin;
    CustomConfig uuid;

    Join(RPG pl) {
        plugin = pl;
        plugin.getServer().getPluginManager().registerEvents(this, plugin);
    }

    @EventHandler(priority = EventPriority.HIGHEST)
    public void onPlayerJoin(PlayerJoinEvent e) {

        uuid = new CustomConfig(plugin, "UUID.yml");
        FileConfiguration config = uuid.getConfig();
        Player p = e.getPlayer();
        UUID id = p.getUniqueId();
        String Job = config.getString("UUID." + id + ".Job");
        String X = config.getString("UUID." + id + ".Spawn.x");
        if (X == null){

            config.set("UUID." + id + ".Spawn.x", -315);
            config.set("UUID." + id + ".Spawn.y", 67);
            config.set("UUID." + id + ".Spawn.z", -444);
            uuid.saveConfig();
        }
        if (Job == null) {
            config.set("UUID." + id + ".Job", "Crafter");


            uuid.saveConfig();
            World w = p.getLocation().getWorld();
            Location spawnpoint = new Location(w, -315, 67, -444);
            p.teleport(spawnpoint);
            // プレイヤーに本を作成して渡す
            ItemStack item = new ItemStack(Material.WRITTEN_BOOK);
            BookMeta meta = (BookMeta) item.getItemMeta();
            meta.setAuthor("yuu_111");
            meta.setDisplayName("チュートリアル");
            meta.addPage(new String[]{
                    // 1ページ目
                    "チュートリアル\n" +
                            "まず最初に奥の方の看板で職業を選びましょう\n" +
                            "各職業についての説明は次のページ以降にあります",
                    // 2ページ目
                    "戦士\n" +
                            "\n" +
                            "aaa\n" +
                            "bbb\n" +
                            "ccc"});
            item.setItemMeta(meta);
            p.getInventory().addItem(item);
        }
    }
}

File: src/main/java/yuu/rpg/Join.kt
package yuu.rpg

import org.bukkit.Location
import org.bukkit.Material
import org.bukkit.World
import org.bukkit.configuration.file.FileConfiguration
import org.bukkit.entity.Player
import org.bukkit.event.EventHandler
import org.bukkit.event.EventPriority
import org.bukkit.event.Listener
import org.bukkit.event.player.PlayerJoinEvent
import org.bukkit.inventory.ItemStack
import org.bukkit.inventory.meta.BookMeta

import java.util.UUID

class Join internal constructor(private val plugin: RPG) : Listener {

    init {
        plugin.server.pluginManager.registerEvents(this, plugin)
    }

    val uuid: CustomConfig = CustomConfig(plugin, "UUID.yml")

    @EventHandler(priority = EventPriority.HIGHEST)
    fun onPlayerJoin(e: PlayerJoinEvent) {

        val config = uuid.getConfig()
        val p = e.player
        val id = p.uniqueId
        val job = config!!.getString("UUID.$id.Job")
        val x = config.getString("UUID.$id.Spawn.x")
        p.sendMessage(x)
        if (x == null) {
            config.set("UUID.$id.Spawn.x", -315)
            config.set("UUID.$id.Spawn.y", 67)
            config.set("UUID.$id.Spawn.z", -444)
            uuid.saveConfig()
        }
        if (job == null) {
            config.set("UUID.$id.Job", "Wanderer")
            uuid.saveConfig()
            val spawnpoint = Location(p.location.world, -315.0, 67.0, -444.0)
            p.teleport(spawnpoint)

            val item = ItemStack(Material.WRITTEN_BOOK)
            val meta = item.itemMeta as BookMeta
            meta.author = "yuu_111"
            meta.displayName = "チュートリアル"
            meta.addPage(
                    "チュートリアル\nまず最初に奥の方の看板で職業を選びましょう\n各職業についての説明は次のページ以降にあります",
                    "戦士\n\n基本的な剣などの武器を扱える職業です\n高い筋力を生かして弓を高い威力で使えます\nccc")
            item.itemMeta = meta
            p.inventory.addItem(item)
        }
        var hp = config.getString("UUID.$id.Lv.$job.HP")
        if (hp == null) {
            config.set("UUID.$id.Lv.$job.HP", 20)
            config.set("UUID.$id.Lv.$job.MP", 20)

        }
    }
}
